narwhal/primary/src/synchronizer.rs

```diff
@@ -1247,49 +1259,36 @@ impl State {
         to_accept
     }
 
-    /// Runs GC on the suspended certificates, returns a list that can be accepted at gc round + 1.
-    /// It is caller's responsibility to check if some children of the returned certificates can
-    /// also be accepted.
-    fn run_gc(&mut self, gc_round: Round) -> Vec<SuspendedCertificate> {
-        // Remove suspended certificates below gc round, and collect digests for certificates just
-        // above the gc round.
-        let mut gc_certificates = Vec::new();
-        let mut certificates_above_gc_round = HashSet::new();
-        while let Some(((round, digest), children)) = self.missing.iter().next() {
-            if *round > gc_round {
-                break;
-            }
-            if *round == gc_round {
-                certificates_above_gc_round.extend(children.iter().cloned());
+    /// Runs GC on the suspended certificates.
+    /// Returns one certificate that can be GC'ed and accepted, or None.
+    ///
+    /// It is the caller's responsibility to check if any children of the returned certificate
+    /// can also be accepted.
+    fn run_gc_once(&mut self, gc_round: Round) -> Option<SuspendedCertificate> {
+        // Accept suspended certificates at and below gc round + 1, because their parents will be
+        // rejected at validation time.
```

Comment:
validation time when they get into the DAG? Adding more info on the comment might help a bit more.